examples/src/bin/composite_template.rs

```diff
@@ -0,0 +1,106 @@
+//! # Composite Template Example
+//!
+//! This sample demonstrates how to create a widget using GTK's composite templates.
+
+#[macro_use]
+extern crate glib;
+#[macro_use]
+extern crate gtk;
+
+use gio::{prelude::ApplicationExtManual, ApplicationExt};
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::{Cast, StaticType};
+use gtk::subclass::prelude::*;
+use gtk::subclass::widget::*;
+use gtk::WidgetExt;
+
+/// The private struct, which can hold widgets and other data.
+#[derive(Debug, CompositeTemplate)]
+pub struct ExApplicationWindowPriv {
```

Comment:
The pattern we use here and elsewhere (see docs) is

```rust
mod imp {
    pub struct ExApplicationWindow {
        [...]
    }

    impl ObjectSubclass for ExApplicationWindow {
        type Type = super::ExApplicationWindow;
    }
}

glib_wrapper! {
    pub struct ExApplicationWindow(ObjectSubclass<imp::ExApplicationWindow>)
        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow, @implements gio::ActionMap, gio::ActionGroup;
}
```

We have proper namespacing in Rust, no need to abuse type names for that :)